internal/kibana/maintenance_window/create.go

```diff
@@ -0,0 +1,65 @@
+package maintenance_window
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/kibana_oapi"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func (r *MaintenanceWindowResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var planMaintenanceWindow MaintenanceWindowModel
+
+	diags := req.Plan.Get(ctx, &planMaintenanceWindow)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Generate API request body from plan
+	body, diags := planMaintenanceWindow.toAPICreateRequest(ctx)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	serverVersion, sdkDiags := r.client.ServerVersion(ctx)
+	if sdkDiags.HasError() {
+		return
+	}
+
+	serverFlavor, sdkDiags := r.client.ServerFlavor(ctx)
+	if sdkDiags.HasError() {
+		return
+	}
+
+	diags = validateMaintenanceWindowServer(serverVersion, serverFlavor)
+	if diags.HasError() {
+		return
+	}
```

Comment:
```suggestion
	isSupported, sdkDiags := r.client.EnforceMinVersion(ctx, version.Must(version.NewVersion("9.1.0")))
	resp.Diagnostics.Append(utils.FrameworkDiagsFromSDK(diags)...)
	if resp.Diagnostics.HasError() {
		return
	}
	
	if !isSupported {
		resp.Diagnostics.AddError("Unsupported server version", "Maintenance windows are not supported until Elastic Stack v9.0. Upgrade the target server to use this resource")
		return
	}
```

EnforceMinVersion will bundle the version/serverless check.